### PR TITLE
Compile CI kernels from Amazon Linux repository

### PR DESCRIPTION
**Pull request by roypat**
Use the amazon linux downstream of the linux kernel to derive our CI
kernels from, as this is closer to an actual production setup than just
taking the tip of upstream releases.
## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by roypat** [comment]
CI Run with new artifacts at https://buildkite.com/firecracker/firecracker-pr/builds/10517#_. I'll rebuild artifacts before merging after I got two approvals.

File: resources/rebuild.sh
Comment (bchalios) [inline resources/rebuild.sh:178]: nit: missing space

File: resources/rebuild.sh
Comment (bchalios) [inline resources/rebuild.sh:181]: let's update this to say something:

```
TODO: drop this (and remove the patches from the repo) when AL backports the
patches to 6.1.
```

File: resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
Comment (bchalios) [inline resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config:311]: How come this goes away? Actually there are some lines like this. Is it because you built the kernel and took the .config?

File: resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
Comment (bchalios) [inline resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config]: That seems wrong. This (and all the rest of ACPI configs) should not be enabled in the previous config for this kernel.

File: resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
Comment (roypat) [reply resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config:311]: For this specific config I used  `make menuconfig` to disable ACPI, which then also removed a bunch of things depending on CONFIG_ACPI.

File: resources/rebuild.sh
Comment (roypat) [reply resources/rebuild.sh:181]: Ah, true, lemme update

File: resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
Comment (roypat) [reply resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config]: I can fold this one back into the previous commit

File: resources/rebuild.sh
Comment (roypat) [reply resources/rebuild.sh]: fixed

File: resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config
Comment (roypat) [reply resources/guest_configs/microvm-kernel-ci-x86_64-5.10-no-acpi.config]: done

File: resources/rebuild.sh
Comment (zulinx86) [inline resources/rebuild.sh:164]: why do we need to call both? isn't `make distclean` enough?
as per [Makefile](https://github.com/torvalds/linux/blob/master/Makefile#L1475), `distclean` depends on `mrproper` and `mrproper` depends on `clean`.

File: resources/rebuild.sh
Comment (zulinx86) [inline resources/rebuild.sh:138]: it seems indentation for this file was 4 spaces. could we keep consistency?

Comment (bchalios) [reply resources/rebuild.sh:138]: <irrelevant-to-PR>that's why we should be using TABs at the beginning of lines</irrelevant-to-PR>

**Review by bchalios — APPROVED** [review]

File: resources/rebuild.sh
Comment (roypat) [reply resources/rebuild.sh:164]: Ah, cool. I think I just saw some instructions at some point in my life that had both, and have been doing both ever since haha

File: resources/rebuild.sh
Comment (roypat) [reply resources/rebuild.sh]: My IDE has seemingly decided that 1 tab = 2 space in .sh files. Anyway, fixed

**Comment by bchalios** [comment]
It seems to be still failing with:

```
E   AssertionError: assert {'aes', 'asim...simdrdm', ...} == {'aes', 'asim...simdrdm', ...}
E     Extra items in the right set:
E     'rng'
E     Full diff:
E       {
E        'aes',
E        'asimd',
E        'asimddp',...
E     
E     ...Full output truncated (27 lines hidden), use '-vv' to show
```

on ARM instances

**Comment by roypat** [comment]
> It seems to be still failing with:
> 
> ```
> E   AssertionError: assert {'aes', 'asim...simdrdm', ...} == {'aes', 'asim...simdrdm', ...}
> E     Extra items in the right set:
> E     'rng'
> E     Full diff:
> E       {
> E        'aes',
> E        'asimd',
> E        'asimddp',...
> E     
> E     ...Full output truncated (27 lines hidden), use '-vv' to show
> ```
> 
> on ARM instances

Yes, that's because it'll only pass once the CI artifacts are actually rebuilt. See https://buildkite.com/firecracker/firecracker-pr/builds/10517# for a run with rebuild CI artifacts where these tests pass.

**Review by bchalios — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Pull request merged**
